### ISSUE TITLE
Cannot use dependency injection in query

### ISSUE THREAD

**Issue by FLXT-Qi** [yarinvak/graphql-spring-annotations#3]
For example, I create a class QueryImpl, as below.
`
@Component
class QueryImpl implements Query {
   @Autowired private DatabaseService db;
}
`

the db is null, spring cannot inject beans into the query.

**Comment by murtuza-ranapur** [comment]
Facing the same issue
`@Component
public class Query implements QueryRoot {
    @Autowired
    private UserService userService;
}`

getting null userService

**Comment by yarinvak** [comment]
Yes i was able to reproduce the error. I am trying to figure out how to solve this. Anyway the better way of handling fields is with data fetchers - create a separate class annotated with component and you can inject the service there

**Comment by xdlv** [comment]
use constructor injection in your classes, do not use field injection.
